### ISSUE TITLE
Bypass SSL certificate verification?

### ISSUE THREAD

**Issue by altjx** [slack-ruby/slack-ruby-client#214]
Is there a way to bypass the SSL certificate verification process from the slack ruby client?

**Comment by dblock** [comment]
Doesn't look possible right now without patching the code. It would probably be different for web and RTM APIs.

For web, we build the Faraday connection for the web client [here](https://github.com/slack-ruby/slack-ruby-client/blob/master/lib/slack/web/faraday/connection.rb). I would start by hacking this code (possibly with [this](https://gist.github.com/doubleotoo/3942158)) to first find out how to disable it there. If you just want to disable peer verification adding `verify: false` into ssl options should do it. 

For RTM you need to make changes to the socket in the appropriate async library, [here](https://github.com/slack-ruby/slack-ruby-client/blob/35ce39808bb7a86f71116cc2cfd0a64b5fb815c7/lib/slack/real_time/concurrency/celluloid.rb#L98) for celluloid.

Once you can figure out what options are needed where we can talk about how to add a feature to configure each scenario externally.

**Comment by altjx** [comment]
Thanks. This might sound ghetto, but as a temporary workaround, I added this to the top of my ruby script:

OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE

Regards,
Alton

Sent from my iPhone

On Aug 13, 2018, at 5:06 PM, Daniel Doubrovkine (dB.) @dblockdotorg <notifications@github.com> wrote:

Doesn't look possible right now without patching the code. It would probably be different for web and RTM APIs.

For web, we build the Faraday connection for the web client here. I would start by hacking this code (possibly with this) to first find out how to disable it there. If you just want to disable peer verification adding verify: false into ssl options should do it.

For RTM you need to make changes to the socket in the appropriate async library, here for celluloid.

Once you can figure out what options are needed where we can talk about how to add a feature to configure each scenario externally.

**Comment by altjx** [comment]
Yup, lol. Thanks man and will do! Thanks for your work on this project. I love it!

Regards,
Alton

Sent from my iPhone

On Aug 13, 2018, at 5:20 PM, Daniel Doubrovkine (dB.) @dblockdotorg <notifications@github.com> wrote:

Aha, that's the underlying library and seems to do it. Be careful though because that disables it globally so YMMV.
